namer/namer.cc

```diff
@@ -1414,6 +1414,57 @@ class SymbolDefiner {
         }
     }
 
+    void deleteViaFullNameHash(core::MutableContext ctx, const core::FoundFieldHash &oldDefHash) {
+        auto ownerRef = core::FoundDefinitionRef(core::FoundDefinitionRef::Kind::Class, oldDefHash.owner.idx);
+        ENFORCE(oldDefHash.nameHash.isDefined(), "Can't delete rename if old hash is not defined");
+
+        // Because a change to classes would have take the slow path, should be safe
+        // to look up old owner in current foundDefs.
+        auto ownerSymbol = getOwnerSymbol(ownerRef);
+        ENFORCE(ownerSymbol.isClassOrModule());
+        auto owner = ownerSymbol.asClassOrModuleRef();
+        if (oldDefHash.owner.onSingletonClass) {
+            owner = owner.data(ctx)->singletonClass(ctx);
+        }
+
+        // We have to accumulate a list of fields to delete, instead of deleting them in the loop
+        // below, because deleting a field invalidates the members() iterator.
+        vector<core::FieldRef> toDelete;
```

Comment:
I don't have strong opinions here, but just a thought: i wonder whether it makes sense to keep a `vector<core::SymbolRef>` here, let the individual refs automatically be upcast on assignment, and then downcast them when needed later for sake of deleting?

Reply:
Slight preference for stronger source-level typing if we are going to do per-symbol-type deletes; if we decided to roll everything up into a single `deleteViaFullNameHash`, then we'd almost have to go the `vector<core::SymbolRef>` route.